internal/engine/buildcontrol/docker_compose_build_and_deployer.go

```diff
@@ -166,8 +166,17 @@ func (bd *DockerComposeBuildAndDeployer) BuildAndDeploy(ctx context.Context, st
 			return store.ImageBuildResult{}, fmt.Errorf("Not an image target: %T", target)
 		}
 
+		var cmd v1alpha1.Cmd
+		if iTarget.CmdImageName != "" {
```

Comment:
`cmd` is going to always be non-nil, which might be unexpected within `Build()` - might be better to pass a nil pointer if there is no custom build command